part/node.go

```diff
@@ -0,0 +1,341 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package part
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+	"unsafe"
+)
+
+type nodeKind uint8
+
+const (
+	nodeKindUnknown = iota
+	nodeKind0
```

Comment:
What is the point of nodeKind0? It doesn't exists in the paper, and is never set. Any non initialized nodes will be 0 / nodeKindUnknown, so there is also no implicit way for it to exist. Can't we just get rid of it?

Reply:
It's used by the leaf nodes. I'll see about removing it.